aws/resource_aws_cloudwatch_event_bus_test.go

```diff
@@ -0,0 +1,186 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_cloudwatch_event_bus", &resource.Sweeper{
+		Name: "aws_cloudwatch_event_bus",
+		F:    testSweepCloudWatchEventBuses,
+		Dependencies: []string{
+			"aws_cloudwatch_event_rule",
+			"aws_cloudwatch_event_target",
+		},
+	})
+}
+
+func testSweepCloudWatchEventBuses(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).cloudwatcheventsconn
+
+	input := &cloudwatchevents.ListEventBusesInput{}
+
+	for {
+		output, err := conn.ListEventBuses(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping CloudWatch Event Bus sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving CloudWatch Event Buses: %s", err)
+		}
+
+		if len(output.EventBuses) == 0 {
+			log.Print("[DEBUG] No CloudWatch Event Buses to sweep")
+			return nil
+		}
+
+		for _, eventBus := range output.EventBuses {
+			name := aws.StringValue(eventBus.Name)
+			if name == "default" {
+				continue
+			}
+
+			log.Printf("[INFO] Deleting CloudWatch Event Bus %s", name)
+			_, err := conn.DeleteEventBus(&cloudwatchevents.DeleteEventBusInput{
+				Name: aws.String(name),
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting CloudWatch Event Bus %s: %s", name, err)
+			}
+		}
+
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+	}
+
+	return nil
+}
+
+func TestAccAWSCloudWatchEventBus_basic(t *testing.T) {
+	var eventBusOutput cloudwatchevents.DescribeEventBusOutput
+	busName := acctest.RandString(15)
```

Comment:
Minor thing: Prefer `acctest.RandomWithPrefix("tf-acc-test")` so that we can easily distinguish provider acceptance resources in the AWS console.